### ISSUE TITLE
add checks for aws_account_id

### ISSUE THREAD

**Issue by Elmo33** [Altinity/terraform-provider-altinitycloud#91]
Tried provisioning env with an empty string, and integer set to aws_account_id and terraform apply was successful.

**Comment by ianaya89** [comment]
Resolved d924ab2

Add regex to validate AWS Account ID. When setting invalid ID, you will see this error message:

```
│ Error: Invalid Attribute Value Match
│
│   with altinitycloud_env_aws.this,
│   on main.tf line 5, in resource "altinitycloud_env_aws" "this":
│    5:   aws_account_id = ""
│
│ Attribute aws_account_id must be a 12-digit number, got:
```

**Comment by Elmo33** [comment]
still an issue
config:
```
terraform {
  required_providers {
    altinitycloud = {
      source = "altinity/altinitycloud"
      version = "0.2.8"
    }
  }
}

provider "aws" {
  region = "us-west-2"
}

provider "altinitycloud" {
  api_url = "https://anywhere.dev.altinity.cloud"
}

resource "altinitycloud_env_certificate" "this" {
  env_name = "altinity-byoc-empty-aws-account-id-a93384fe-a"
}

module "altinitycloud_connect_aws" {
  source = "altinity/connect-aws/altinitycloud"
  pem = altinitycloud_env_certificate.this.pem
  url = "https://anywhere.dev.altinity.cloud"
}

resource "altinitycloud_env_aws" "this" {
  name = "altinity-byoc-empty-aws-account-id-a93384fe-a"
  cidr = "10.1.0.0/21"
  region = "us-west-2"
  aws_account_id = ""
  zones = ["us-west-2a", "us-west-2b"]
  load_balancers = {"public": {"enabled": true}}
  node_groups = [{"name": "t4g.large", "node_type": "t4g.large", "capacity_per_zone": 10, "reservations": ["SYSTEM"]}]
  cloud_connect = true
  force_destroy = true
  force_destroy_clusters = true
  depends_on = [module.altinitycloud_connect_aws]
}

```
